advanced-usage/future-mocking.md

```diff
@@ -1,619 +1,513 @@
-
-  {{region FutureMocking#SampleCode5}}
-    Public Sub service_Saved(ByVal sender As Object, ByVal e As EntrySavedEventArgs)
-        Me.label1.Text = "Saved string : " & e.EntryValue
-    End Sub
-  {{endregion}}
-
-Next, is a simple test using [MSpec](https://codebetter.com/aaronjensen/2008/05/08/introducing-machine-specifications-or-mspec-for-short/) where we have created an event handler that will receive as an argument the value that was passed to the `service.Save` call in the `SaveToDatabase` method.
-
-  #### __[C#]__
-  
-  {{region FutureMocking#SampleCode6}}
-    [Subject(typeof(Form))]
-    public class when_save_to_database_is_invoked_on_form
-    {
-        Establish context = () =>
-        {
-            IEntryService serviceMock = Mock.Create<EntryService>();
-            Mock.Arrange(() => serviceMock.Save(valueToSave)).Raises(() => serviceMock.Saved += null, new EntrySavedEventArgs(valueToSave));
-        };
-    
-        private Because of = () =>
-        {
-            sut = new Form();
-            sut.SaveToDatabase(valueToSave);
-        };
-    
-        private It should_assert_that_label_contains_expected_valueToSave = () =>
-            sut.label1.Text.ShouldEqual("Saved string : " + valueToSave);
-    
-        static Form sut;
-        const string valueToSave = "Raise Event";
-    }
-  {{endregion}}
-
-  #### __[VB]__
-
-  {{region FutureMocking#SampleCode6}}
-    <Subject(GetType(Form))>
-    Public Class when_save_to_database_is_invoked_on_form
-        Private context As Establish = Function()
-                                          Dim serviceMock As IEntryService = Mock.Create(Of EntryService)()
-                                          Mock.Arrange(Function() serviceMock.Save(valueToSave)).Raises(Function() CSharpImpl.__Assign(serviceMock.Saved, Nothing), New EntrySavedEventArgs(valueToSave))
-                                      End Function
-
-        Private [of] As Because = Function()
-                                      sut = New Form()
-                                      sut.SaveToDatabase(valueToSave)
-                                  End Function
-
-        Private should_assert_that_label_contains_expected_valueToSave As It = Function() sut.label1.Text.ShouldEqual("Saved string : " & valueToSave)
-        Shared sut As Form
-        Const valueToSave As String = "Raise Event"
-
-        Private Class CSharpImpl
-            <Obsolete("Please refactor calling code to use normal Visual Basic assignment")>
-            Shared Function __Assign(Of T)(ByRef target As T, value As T) As T
-                target = value
-                Return value
-            End Function
-        End Class
-    End Class
-  {{endregion}}
-
-Here we can see that although no instance is supplied to the target UI class, JustMock picks up the intended setup from the context.
-
-## See Also
-
- * [Fluent Mocking]({%slug justmock/basic-usage/fluent-mocking%})
+---
+title: Working with Instances
```

Comment:
I don't like the `Future Mocking` term as it does not provide an instant understanding of what this functionality and article are about. But I prefer it in comparison to the suggested `Working with Instances` especially in terms of SEO.